File: stack/剑指 Offer 27. 二叉树的镜像.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
 var mirrorTree = function(root) {
    // 解题思路： 递归交换左右子树

    // 如果为空树或者为唯一节点数--直接返回
    if(root == null || (root.left == null && root.right == null)) return root;

    // 交换左右子树 es6 语法
    [root.left,root.right] = [root.right,root.left];
    // 递归调用交换左右子树的左右子树
    mirrorTree(root.left);
    mirrorTree(root.right);
    // 返回根节点。
    return root;
};

File: Python/剑指 Offer 27. 二叉树的镜像.py
# -*- codeing = utf-8 -*-
# @Time : 2023-8-20 17:48
# @Author : 刘永奇
# @File : 剑指 Offer 27. 二叉树的镜像.py
# @Software : PyCharm
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def mirrorTree(self, root: TreeNode) -> TreeNode:
        self.mirror(root)
        return root

    def mirror(self, root):
        if not root: return
        root.left, root.right = root.right, root.left
        self.mirror(root.left)
        self.mirror(root.right)